src/STAN.Client/BlockingChannel.cs

```diff
@@ -48,14 +48,19 @@ internal ICollection<TKey> Keys
             }
         }
 
-        internal void waitForSpace()
+        internal bool TryWaitForSpace(int millisecondsTimeout)
```

Comment:
I notice the only place this is used the return value is discarded.  Maybe remove the bool rv?

Reply:
I've reworked it to no longer return a boolean, and also to no longer loop. It did not make a difference in terms of deadlocks, and throughput slightly increased without the loop (1350 msg/sec -> ~1500 msg/sec; 100 max in flight).